src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/Converters/TypeConverter.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Mcp.Converters;
+
+internal class TypeConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        // Only expecting a single argument, otherwise use a different converter
+        if (!context.FunctionContext.TryGetToolInvocationContext(out ToolInvocationContext? toolContext)
+            || toolContext.Arguments?.Count != 1)
+        {
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        object? argValue = toolContext.Arguments.Values.FirstOrDefault();
```

Comment:
What does this mean? Why would we limit this to only handling the case of a single argument?

Reply:
I was about to leave a similar question here. Can you please describe the scenario?